upstairs/src/lib.rs

```diff
@@ -1637,9 +1650,13 @@ impl Buffer {
         assert!(offset + data.len() <= self.data.len());
         assert_eq!(offset % self.block_size, 0);
         assert_eq!(data.len() % self.block_size, 0);
-        for (i, d) in data.iter_mut().enumerate() {
-            if self.owned[offset + i] {
-                *d = self.data[offset + i];
+
+        let start_block = offset / self.block_size;
+        for (b, chunk) in data.chunks_mut(self.block_size).enumerate() {
+            assert_eq!(chunk.len(), self.block_size);
```

Comment:
A little redundant with the `data.len() % blocksize` assertion above?  Make make debug_assert_eq?

Reply:
Yeah, this assertion is intended as executable documentation more than an actual check, so making it a debug assertion seems fine.  Done in 0b24644d8678fcfcdac84d8f8d0cc9da9af0d146